plugin/src/main/java/org/wildfly/plugin/provision/ApplicationImageMojo.java

```diff
@@ -0,0 +1,223 @@
+     *     If the image is pushed and the registry is not set, it defaults to "docker.io" to login to the registry
+     *     --&gt;
+     *   &lt;registry&gt;quay.io&lt;/registry&gt;
+     *
+     *   &lt;!-- (optional) The user name to login to the container registry (if push is enabled). --&gt;
+     *   &lt;user&gt;${user.name}&lt;/user&gt;
+     *
+     *   &lt;!-- (optional) The password login to the container registry (if push is enabled) --&gt;
+     *   &lt;password&gt;${my.secret.password}&lt;/password&gt;
+     * &lt;/image&gt;
+     * </pre>
+     */
+    @Parameter(alias = "image")
+    private ApplicationImageInfo image;
+
+    @Override
+    protected String getGoal() {
+        return "image";
+    }
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        super.execute();
+
+        if (image == null) {
+            image = new ApplicationImageInfo();
+        }
+
+        try {
+            // The Dockerfile is always generated when the image goal is run.
+            // This allows the user to then use the generated Dockerfile in other contexts than Maven.
+            String runtimeImage = this.image.getWildFlyRuntimeImage();
+            getLog().info(format("Generating Dockerfile %s from base image %s",
+                    Paths.get(project.getBuild().getDirectory()).resolve("Dockerfile"),
+                    runtimeImage));
+            generateDockerfile(runtimeImage, Paths.get(project.getBuild().getDirectory()), provisioningDir);
+
+            if (!image.build) {
+                return;
+            }
+
+            if (!isDockerBinaryAvailable(image.dockerBinary)) {
+                throw new MojoExecutionException(String.format("Unable to build application image with %s. Please check your %s installation",
+                        image.dockerBinary, image.dockerBinary));
+            }
+
+            String image = this.image.getApplicationImageName(project.getArtifactId());
+
+            boolean buildSuccess = buildApplicationImage(image, runtimeImage);
+            if (!buildSuccess) {
+                throw new MojoExecutionException(String.format("Unable to build application image %s", image));
+            }
+            getLog().info(String.format("Successfully built application image %s", image));
+
+            if (this.image.push) {
+                logToRegistry();
+
+                boolean pushSuccess = pushApplicationImage(image);
+                if (!pushSuccess) {
+                    throw new MojoExecutionException(String.format("Unable to push application image %s", image));
+                }
+                getLog().info(String.format("Successfully pushed application image %s", image));
+            }
+        } catch (IOException e) {
+            MojoExecutionException ex = new MojoExecutionException(e.getLocalizedMessage());
+            ex.initCause(e);
+            throw ex;
```

Comment:
Couldn't we just use the `new MojoExecutionException(e.getLocalizedMessage(), e);`?

Reply:
fixed, thanks